Assignment3

```diff
@@ -0,0 +1,60 @@
+
+1. Is the following code valid?  If so, what information or assumptions must be given for it to be valid?  If not, why not?
+	```c 
+	for(;i < 1000;i*=2) 
+	{
+	    printf(“%d\n”);
+	}
+	  ```
+Answer:
+No. Because the initialization is lost in the for function, and there is no value after the "%d\n" as well.
+
+2. What, if anything, is wrong with the following code fragment?  You may assume all variables are defined.
+	```c 
+	for(i=0;i<10;i--);
+	{
+		i=(int) (i*0.5);
+	}
+	```
+Answer:
+0.5 is not an integer, and probably i*0.5 is not an integer as well.
```

Comment:
almost! what would happen because of this is it would enter an infinite loop. also there's a semi colon that would cause a compile warning.